ocrd_lib.sh

```diff
@@ -29,13 +36,25 @@ init() {
     exit 2
   fi
 
+  WORKDIR=ocr-d/"$PROCESS_DIR" # use subdirectory of same volume so --reflink CoW still possible
+  if ! mkdir -p "$WORKDIR"; then
+    logger -p user.error -t $TASK "insufficient permissions on /data volume"
+    exit 5
+  fi
+  # try to be unique here (to avoid clashes)
+  REMOTEDIR="KitodoJob_${PID}_$(basename $PROCESS_DIR)"
+
   WORKFLOW=$(command -v "$WORKFLOW" || realpath "$WORKFLOW")
   if ! test -f "$WORKFLOW"; then
     logger -p user.error -t $TASK "invalid workflow '$WORKFLOW'"
     exit 3
   fi
   logger -p user.notice -t $TASK "using workflow '$WORKFLOW':"
   ocrd_format_workflow | logger -p user.notice -t $TASK
+  if test "${WORKFLOW#/workflows/}" = "$WORKFLOW"; then
```

Comment:
I do not understand the check here cause of  [Line 48](https://github.com/slub/ocrd_manager/blob/07227b5192d18a63afab299b33cc7e3b6b740095/ocrd_lib.sh#L48) and i think that needs a comment here. 

So with this parameter substitution it looks like the file test Line 48 will fail or this check will not be successful. 

IMO for example:

With /workflows/ocr-workflow-default.sh as `$WORKFLOW` Line 48 will successful.

Line 54 `"ocr-workflow-default.sh" = "/workflows/ocr-workflow-default.sh"` will fail.

Reply:
That's precisely the point. The check trivially looks whether the realpath of the workflow starts with /workflows and thus, needs no further action. If it does not, then it gets copied into the workdir under a default name, to ensure it will be accessible.

Reply:
Alone that already the inquiry came from me would be worth it to write comment there cause it is not easy to read (for none linux power users those for whom parameter substitution syntax has not yet become flesh and blood :wink: ) and understand why you do that here.

Reply:
The problem is that by this logic, I would have to second every other statement in the script with an explanatory comment. Code is also documentation in itself. Only complex / non-trivial code bits should be commented.

If you encounter an unknown construct in a (new) language, it's up to you to visit the documentation first. This expansion test is very common in shell scripts.

Feel free to make a suggestion with a better formulation (preferbly _inside_ the condition, i.e. from the perspective after the test succeeded).

Reply:
> Feel free to make a suggestion with a better formulation (preferbly _inside_ the condition, i.e. from the perspective after the test succeeded).

Maybe a little bit simpler otherwise i would separate the variable or leave a small comment.
```suggestion
  if test "${WORKFLOW:0:11}" != "/workflow/"; then
```
IMO not only knowledge was required, you also had to think outside the box. I think code should be clearly visible without thinking outside the box. But I understand your points and I can also live with the current state.

Reply:
I don't know what you mean by thinking outside the box here. This substitution test is straightforward and very common (e.g. you can see tons of these tests in every autoconf script). Your offeset-length test would also work (with plural `s`), but is less idiomatic.

I just added a comment – happy now?

(You still need to unrequest changes, otherwise the PR is still blocked.)

Reply:
Thx sorry for the circumstances :zipper_mouth_face: